questionnaire/questionnaire-service.js

```diff
@@ -448,6 +449,31 @@ function createQuestionnaireService({
     async function getProgressEntries(questionnaireId, query) {
         // 1 - get questionnaire instance
         const questionnaire = await getQuestionnaire(questionnaireId);
+        // 1.1 check questionnaire is a compatible version
+        const isCompatible = checkQuestionnaireVersion(questionnaire.version);
```

Comment:
checkQuestionnaireVersion is not too descriptive of it's purpose here.

Have a function (util) with a sole purpose of checking if `isQuestionnaireVersionCompatible`. This util would then employ your `getInstalledModuleVersion` util and the `versionComparison` util.

````js
const isCompatible = isQuestionnaireVersionCompatible(questionnaire.version)// questionnaire instance version.

could be a case of a simple rename if I understand this fully.